src/ds3/models/testParsers.go

```diff
@@ -0,0 +1,372 @@
+}
+
+func (aggregateError *AggregateError) Append(err error) {
+    if err != nil {
+        aggregateError.Errors = append(aggregateError.Errors, err)
+    }
+}
+
+//TODO test parsers- hand implemented
+
+func (jobList *JobList) parse(node *XmlNode, aggErr *AggregateError) {
+    // No attributes to parse
+
+    // Parse children nodes
+    for _, child := range node.Children {
+        switch child.XMLName.Local {
+        case "Job":
+            var job Job
+            job.parse(&child, aggErr)
+            jobList.Jobs = append(jobList.Jobs, job)
+        }
+    }
+}
+
+func (job *Job) parse(node *XmlNode, aggErr *AggregateError) {
+    // Parse attributes
+    for _, attr := range node.Attrs {
+        switch attr.Name.Local {
+        case "Aggregating":
+            job.Aggregating = parseBoolFromString(attr.Value, aggErr)
+        case "BucketName":
+            job.BucketName = parseNullableStringFromString(attr.Value)
+        case "CachedSizeInBytes":
+            job.CachedSizeInBytes = parseInt64FromString(attr.Value, aggErr)
+        case "ChunkClientProcessingOrderGuarantee":
+            parseEnumFromString(attr.Value, &job.ChunkClientProcessingOrderGuarantee, aggErr)
+        case "CompletedSizeInBytes":
+            job.CompletedSizeInBytes = parseInt64FromString(attr.Value, aggErr)
+        case "EntirelyInCache":
+            job.EntirelyInCache = parseBoolFromString(attr.Value, aggErr)
+        case "JobId":
+            job.JobId = attr.Value
+        case "Naked":
+            job.Naked = parseBoolFromString(attr.Value, aggErr)
+        case "Name":
+            job.Name = parseNullableStringFromString(attr.Value)
+        case "OriginalSizeInBytes":
+            job.OriginalSizeInBytes = parseInt64FromString(attr.Value, aggErr)
+        case "Priority":
+            parseEnumFromString(attr.Value, &job.Priority, aggErr)
+        case "RequestType":
+            parseEnumFromString(attr.Value, &job.RequestType, aggErr)
+        case "StartDate":
+            job.StartDate = attr.Value
+        case "Status":
+            parseEnumFromString(attr.Value, &job.Status, aggErr)
+        case "UserId":
+            job.UserId = attr.Value
+        case "UserName":
+            job.UserName = parseNullableStringFromString(attr.Value)
+        }
+    }
+
+    // Parse children nodes
+    for _, child := range node.Children {
+        switch child.XMLName.Local {
+        case "Nodes":
+            job.Nodes = parseJobNodeSlice("Node", child.Children, aggErr)
+        }
+    }
+}
+
+func parseJobNodeSlice(tagName string, xmlNodes []XmlNode, aggErr *AggregateError) []JobNode {
+    var result []JobNode
+    for _, curXmlNode := range xmlNodes {
+        if curXmlNode.XMLName.Local == tagName {
+            var curResult JobNode
+            curResult.parse(&curXmlNode, aggErr)
+            result = append(result, curResult)
+        } else {
+            // TODO possibly change from logging to appending to aggregate error
```

Comment:
If this really is a warning, I wouldn't lump it in with errors.  Doing so would ambiguate the meaning of error.